src/libraries/System.Formats.Tar/tests/TarTestsBase.cs

```diff
@@ -299,6 +299,7 @@ protected void SetCommonSymbolicLink(TarEntry symbolicLink)
 
             // LinkName
             Assert.Equal(DefaultLinkName, symbolicLink.LinkName);
+            Assert.Throws<ArgumentNullException>(() => symbolicLink.LinkName = null);
```

Comment:
Is empty string valid? This code seems to disallow it:
https://github.com/dotnet/runtime/blob/51191575ccced7759ab2b82c7348a488f56d6f52/src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs#L424-L427

Reply:
Yes, it's valid because it's the default value returned by LinkName for all other entry types that are not hard links or symbolic links, even if internally we represent it with a nullable string and the default value is null.

That code you linked doesn't explicitly disallow it, just skips writing it if the value is null or empty. The disallow occurrs in the setter.

Reply:
What does a hardlink or symlink entry with empty LinkName mean?

Reply:
As we discussed offline, empty string is also invalid. This also sprout a bug on TarWriter https://github.com/dotnet/runtime/issues/74887.